services/motion/builtin/move_request.go

```diff
@@ -682,11 +796,46 @@ func (ms *builtIn) createBaseMoveRequest(
 		return nil, err
 	}
 
-	startPoseIF, err := kb.CurrentPosition(ctx)
+	planningFS := baseOnlyFS
+	_, ok := kinematicFrame.(tpspace.PTGProvider)
+	if ok {
```

Comment:
This seems off to me, we're not actually using the fact that the kinematicFrame is a PTGProvider, which is the reason why we would want to cast it, same thing as the other place in the file where we do it.  If there is nothing special about it being a PTG provider we shouldnt have to case it

Reply:
Let's say we've entered `WrapWithKinematics` and then went into `wrapWithDifferentialDriveKinematics` this creates a planningFrame with `New2DMobileModelFrame`.
For this type of frame, inputs place us where we are in the world, so we do not need to do anything special for the `planningFS`.


Conversely, lets say we've entered `WrapWithKinematics` and then went into `wrapWithPTGKinematics` this creates a planningFrame with `NewPTGFrameFromKinematicOptions`.
For this type of frame, inputs place us relative to the origin and not where we are in the world.
We then need to do some editing to the `planningFS`. Specifically, the sole edit is adding a transform frame upstream of the kinematics frame so we can position ourselves absolutely. this is why we do `_, ok := kinematicFrame.(tpspace.PTGProvider)`.

This is also why we import `"go.viam.com/rdk/motionplan/tpspace"`